src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerActor.scala

```diff
@@ -132,8 +132,8 @@ private[impl] class GroupManagerActor(
         plan = DeploymentPlan(from, to, resolve, version, toKill)
         _ = validateOrThrow(plan)(DeploymentPlan.deploymentPlanValidator(config))
         _ = log.info(s"Computed new deployment plan:\n$plan")
-        _ <- scheduler.deploy(plan, force)
         _ <- groupRepo.storeRoot(plan.target, plan.createdOrUpdatedApps, plan.deletedApps)
+        _ <- scheduler.deploy(plan, force)
```

Comment:
The order is definitively important here.
I would say the order was correct before and is wrong now, since a deployment can be rejected. This is most usually the case, if there is a deployment that affects the same app (no force). In that case the future will fail.
We should store the group only, if a deployment is started.

What was the reasoning to change the order?